Entity load error

```
app-1  | [Nest] 225  - 07/02/2024, 12:55:32 PM   ERROR [ExceptionHandler] Cannot use import statement outside a module
app-1  | /app/src/customer/customer.entity.ts:1
app-1  | import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm'
```
Missing datasource import
```
app-1  | src/app.module.ts:30:35 - error TS2304: Cannot find name 'DataSource'.
app-1  | 
app-1  | 30   constructor(private dataSource: DataSource) {}
```

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,16 @@
 run-development-server:
 	@echo "Running development server at http://localhost:3000"
-	docker compose up -d
+	docker compose up -d && \
 	docker compose exec app npm run migration:run
+
+lint:
+	@echo "Linting the code"
+	docker compose exec app npm run lint
+
+test-cov:
+	@echo "Running tests with coverage"
+	docker compose exec app npm run test:cov
+
+test: dist/
+	@echo "Running tests"
+	docker compose exec app npm run test

--- a/README.md
+++ b/README.md
@@ -22,14 +22,17 @@ make run-development-server
 npm run start:prod
 ```
 
-## Test
+## Test & Lint
 
 ```bash
-# unit tests (TODO)
-$ npm run test
+# lint
+$ make lint
 
-# test coverage (TODO)
-$ npm run test:cov
+# unit tests
+$ make test
+
+# test coverage
+$ make test-cov
 ```
 
 ## Migrations

--- a/src/app.controller.spec.ts
+++ b/src/app.controller.spec.ts
@@ -1,22 +1,22 @@
-import { Test, TestingModule } from '@nestjs/testing';
-import { AppController } from './app.controller';
-import { AppService } from './app.service';
+import { Test, TestingModule } from '@nestjs/testing'
+import { AppController } from './app.controller'
+import { AppService } from './app.service'
 
 describe('AppController', () => {
-  let appController: AppController;
+  let appController: AppController
 
   beforeEach(async () => {
     const app: TestingModule = await Test.createTestingModule({
       controllers: [AppController],
       providers: [AppService],
-    }).compile();
+    }).compile()
 
-    appController = app.get<AppController>(AppController);
-  });
+    appController = app.get<AppController>(AppController)
+  })
 
   describe('root', () => {
     it('should return "Hello World!"', () => {
-      expect(appController.getHello()).toBe('Hello World!');
-    });
-  });
-});
+      expect(appController.getHello()).toBe('Hello World!')
+    })
+  })
+})

--- a/src/app.controller.ts
+++ b/src/app.controller.ts
@@ -1,5 +1,5 @@
-import { Controller, Get } from '@nestjs/common';
-import { AppService } from './app.service';
+import { Controller, Get } from '@nestjs/common'
+import { AppService } from './app.service'
 
 @Controller()
 export class AppController {
@@ -7,6 +7,6 @@ export class AppController {
 
   @Get()
   getHello(): string {
-    return this.appService.getHello();
+    return this.appService.getHello()
   }
 }

--- a/src/app.module.ts
+++ b/src/app.module.ts
@@ -1,6 +1,7 @@
 import { Module } from '@nestjs/common'
 import { TypeOrmModule } from '@nestjs/typeorm'
 import { ConfigModule, ConfigService } from '@nestjs/config'
+import { DataSource } from 'typeorm'
 
 import { AppController } from './app.controller'
 import { AppService } from './app.service'

--- a/src/app.service.ts
+++ b/src/app.service.ts
@@ -1,8 +1,8 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable } from '@nestjs/common'
 
 @Injectable()
 export class AppService {
   getHello(): string {
-    return 'Hello World!';
+    return 'Hello World!'
   }
 }

--- a/src/config/typeorm.ts
+++ b/src/config/typeorm.ts
@@ -10,8 +10,8 @@ const datasourceOptions: DataSourceOptions = {
   username: process.env.MYSQL_USER,
   password: process.env.MYSQL_PASSWORD,
   database: process.env.MYSQL_DATABASE,
-  entities: ['dist/src/**/*.entity{.ts,.js}'],
-  migrations: ['dist/src/migrations/*{.ts,.js}'],
+  entities: ['dist/**/*.entity{.ts,.js}'],
+  migrations: ['dist/migrations/*{.ts,.js}'],
   migrationsTableName: 'migrations_typeorm',
 }
 

--- a/src/config/typeorm.ts
+++ b/src/config/typeorm.ts
@@ -10,8 +10,8 @@ const datasourceOptions: DataSourceOptions = {
   username: process.env.MYSQL_USER,
   password: process.env.MYSQL_PASSWORD,
   database: process.env.MYSQL_DATABASE,
-  entities: ['src/**/*.entity{.ts,.js}'],
-  migrations: ['src/migrations/*{.ts,.js}'],
+  entities: ['dist/src/**/*.entity{.ts,.js}'],
+  migrations: ['dist/src/migrations/*{.ts,.js}'],
   migrationsTableName: 'migrations_typeorm',
 }
 

--- a/src/customer/customer.service.spec.ts
+++ b/src/customer/customer.service.spec.ts
@@ -1,18 +1,18 @@
-import { Test, TestingModule } from '@nestjs/testing';
-import { CustomerService } from './customer.service';
+import { Test, TestingModule } from '@nestjs/testing'
+import { CustomerService } from './customer.service'
 
 describe('CustomerService', () => {
-  let service: CustomerService;
+  let service: CustomerService
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
       providers: [CustomerService],
-    }).compile();
+    }).compile()
 
-    service = module.get<CustomerService>(CustomerService);
-  });
+    service = module.get<CustomerService>(CustomerService)
+  })
 
   it('should be defined', () => {
-    expect(service).toBeDefined();
-  });
-});
+    expect(service).toBeDefined()
+  })
+})

--- a/src/migrations/1719924445147-CustomerTable.ts
+++ b/src/migrations/1719924445147-CustomerTable.ts
@@ -1,22 +1,21 @@
-import { MigrationInterface, QueryRunner } from "typeorm";
+import { MigrationInterface, QueryRunner } from 'typeorm'
 
 export class CustomerTable1719924445147 implements MigrationInterface {
-    name = 'CustomerTable1719924445147'
+  name = 'CustomerTable1719924445147'
 
-    public async up(queryRunner: QueryRunner): Promise<void> {
-        await queryRunner.query(`
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
             CREATE TABLE \`customer\` (
                 \`id\` int NOT NULL AUTO_INCREMENT,
                 \`name\` varchar(255) NOT NULL,
                 PRIMARY KEY (\`id\`)
             ) ENGINE = InnoDB
-        `);
-    }
+        `)
+  }
 
-    public async down(queryRunner: QueryRunner): Promise<void> {
-        await queryRunner.query(`
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
             DROP TABLE \`customer\`
-        `);
-    }
-
+        `)
+  }
 }

--- a/test/app.e2e-spec.ts
+++ b/test/app.e2e-spec.ts
@@ -1,24 +1,24 @@
-import { Test, TestingModule } from '@nestjs/testing';
-import { INestApplication } from '@nestjs/common';
-import * as request from 'supertest';
-import { AppModule } from './../src/app.module';
+import { Test, TestingModule } from '@nestjs/testing'
+import { INestApplication } from '@nestjs/common'
+import * as request from 'supertest'
+import { AppModule } from './../src/app.module'
 
 describe('AppController (e2e)', () => {
-  let app: INestApplication;
+  let app: INestApplication
 
   beforeEach(async () => {
     const moduleFixture: TestingModule = await Test.createTestingModule({
       imports: [AppModule],
-    }).compile();
+    }).compile()
 
-    app = moduleFixture.createNestApplication();
-    await app.init();
-  });
+    app = moduleFixture.createNestApplication()
+    await app.init()
+  })
 
   it('/ (GET)', () => {
     return request(app.getHttpServer())
       .get('/')
       .expect(200)
-      .expect('Hello World!');
-  });
-});
+      .expect('Hello World!')
+  })
+})

